### PR TITLE
fix: transition to auth display

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Fix the transition issue by waiting for the next tick before displaying the new UI.
- This ensures the window is opened.
- I also tried making the `onProviderClick` an async function that does the next tick but it might impact other areas, with that, I only kept the changes within the primary area of concern.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-175 #done

### CODE REVIEW

File: packages/shared/src/components/auth/AuthOptions.tsx [hunk 1]
```diff
@@ -421,8 +422,9 @@ function AuthOptions({
                 email: existingEmail,
               });
             }}
-            onProviderClick={(provider, login) => {
+            onProviderClick={async (provider, login) => {
               onProviderClick(provider, login);
+              await nextTick();
```
Comment (capJavert) [inline packages/shared/src/components/auth/AuthOptions.tsx:427]: Waiting for oauth window?

Comment (capJavert) [reply packages/shared/src/components/auth/AuthOptions.tsx:427]: Gotcha, it won't be perfect due to nature of timers but worth a try.

**Review by capJavert — APPROVED** [review]

**Review by rebelchris — APPROVED** [review]
Please do give it a try in prod :)

**Review by nensidosari — APPROVED** [review]

**Comment by sshanzel** [comment]
> Please do give it a try in prod :)

Tested, and things are still working as intended. Hopefully, on other ends too.

**Pull request merged**